File: service-common/src/main/java/util/UtilNetty.java
package util;

import com.google.gson.Gson;
import io.netty.handler.codec.http.websocketx.TextWebSocketFrame;
import lombok.SneakyThrows;
import lombok.val;
import packages.BridgePackage;
import packages.PackageWrapper;

/**
 * @author Рейдж 03.10.2021
 * @project ThePit
 */
public class UtilNetty {

    private static final Gson gson = new Gson();

    public static TextWebSocketFrame toFrame(BridgePackage thePitPackage) {
        return new TextWebSocketFrame(gson.toJson(new PackageWrapper(thePitPackage.getClass().getName(), gson.toJson(thePitPackage))));
    }

    @SneakyThrows
    public static BridgePackage readFrame(TextWebSocketFrame textFrame) {
        PackageWrapper wrapper = gson.fromJson(textFrame.text(), PackageWrapper.class);
        return (BridgePackage) gson.fromJson(wrapper.getObjectData(), Class.forName(wrapper.getClazz()));
    }

}

File: common/src/main/kotlin/me/reidj/common/util/UtilNetty.kt
package me.reidj.common.util

import com.google.gson.Gson
import io.netty.handler.codec.http.websocketx.TextWebSocketFrame
import me.reidj.common.`package`.BridgePackage
import me.reidj.common.`package`.PackageWrapper

object UtilNetty {

    private val gson = Gson()

    fun toFrame(thePitPackage: BridgePackage): TextWebSocketFrame? {
        return TextWebSocketFrame(
            gson.toJson(
                PackageWrapper(
                    thePitPackage::class.java.name,
                    gson.toJson(thePitPackage)
                )
            )
        )
    }

    fun readFrame(textFrame: TextWebSocketFrame): BridgePackage {
        val wrapper: PackageWrapper = gson.fromJson(textFrame.text(), PackageWrapper::class.java)
        return gson.fromJson(wrapper.objectData, Class.forName(wrapper.clazz)) as BridgePackage
    }
}